single_tech_samples/storage/sample1_data_lifecycle/README.md

```diff
@@ -0,0 +1,224 @@
+- [Terraform](https://www.terraform.io) installed on the local machine
+  - *Installation instructions* can be found [here](https://developer.hashicorp.com/terraform/downloads)
+  - Minimum required version is 1.3.x or later
+
+### 2.2. Setup and deployment
+
+> **IMPORTANT NOTE:** As with all Azure Deployments, this will **incur associated costs**. Remember to teardown all related resources after use to avoid unnecessary costs. See [here](#4.3.-deployed-resources) for a list of deployed resources.
+
+Below listed are the steps to deploy this sample :
+
+- Fork and clone this repository. Navigate to  `single_tech_samples/storage/sample1_data_lifecycle/terraform`.
+
+- The sample depends on the variables to be set before the deployment script is run; You can set these variables by editing the variables.tf and terraform.auto.tfvars files, otherwise the default values provided would be considered.
+
+The script is configuration driven and based on a variable of type map(map(map))).
+
+- First level map represents storage accounts.
+- Second level map represents the containers inside the respective storage account.
+- Third level map represents the data life cycle rules under respective container.
+
+Sample variable format is shown below:
+
+```json
+storage_account_container_config = {
+    "StorageAccountName1" = {
+        "Container1"       = { "LifeCycleAction1" : "NumberOfDays", "LifeCycleAction2" : "NumberOfDays"}
+    }
+    "StorageAccountName2" = {
+        "Container2"       = { "LifeCycleAction1" : "NumberOfDays", "LifeCycleAction1" : "NumberOfDays" }
+    }
+    "StorageAccountName3" = {
+        "Container3"       = { "LifeCycleAction1" : "NumberOfDays" }
+        "Container4"       = {   }
+    }
+}
+```
+
+Configure as many storage accounts, containers and lifecycle rules as needed and run the IAC script provided in this git repo.
+
+Run below commands in the sequence:
+
+```bash
+terraform init
+terraform plan
+terraform apply
+```
+
+   > Note: The script would expect that you are logged in to the Azure account using az login command, before you deploy resources to Azure.
+
+### 2.3. Deployed Resources
+
+The following resources will be deployed as a part of this sample once the script is executed:
+
+- Azure Storage Accounts.
+- Azure Storage Containers.
+- Azure data lifecycle rules.
+
+### 2.4. Deployment validation and Execution
+
+The following steps can be performed to validate the correct deployment and execution of the sample:
+
+Users with appropriate access rights should be able to:
+
+- View the storage accounts in the Azure portal.
+- View the storage containers under the storage accounts through the Azure portal.
+- View the data lifecycle rules created in the Azure portal under storage accounts.
+
+Here are the detail steps on how to execute the sample:
+
+- Checkout the code.
+- Configure the variables as needed under the files variables.tf and terraform.auto.tfvars
```

Comment:
Add "." at the end.

Reply:
Put variables.tf and terraform.auto.tfvars in backquotes:
```
`variables.tf` and `terraform.auto.tfvars`
```

Reply:
Done